mono/metadata/object.c

```diff
@@ -4010,38 +4010,56 @@ static int num_main_args = 0;
 MonoArray*
 mono_runtime_get_main_args (void)
 {
+	HANDLE_FUNCTION_ENTER ();
 	MONO_REQ_GC_UNSAFE_MODE;
 	MonoError error;
-	MonoArray *result = mono_runtime_get_main_args_checked (&error);
-	mono_error_assert_ok (&error);
-	return result;
+	MonoArray *result = NULL;
+	error_init (&error);
+	MonoArrayHandle arg_array = mono_runtime_get_main_args_handle (&error);
+	goto_if_nok (&error, leave);
+	result = MONO_HANDLE_RAW (arg_array);
+leave:
+	HANDLE_FUNCTION_RETURN_VAL (result);
```

Comment:
This could be `HANDLE_FUNCTION_RETURN_OBJ` - it's the same behavior, but semantically it means you're returning a raw ptr to a managed object, as opposed to a value.